### DISCUSSION TITLE
How to allow different replication for different volume tier

### DISCUSSION THREAD

**Discussion by NyaMisty** [seaweedfs/seaweedfs#5332] [Q&A] [answered]
Let's say I got a path `/buckets/test-bucket`, and I setup its replication strategy into **001**, and storing it into **hdd** using fs.configure like this:
```
{
  "locationPrefix":  "/buckets/test-buckett",
  "collection":  "",
  "replication":  "001",
  "ttl":  "",
  "diskType":  "hdd",
  "fsync":  false,
  "volumeGrowthCount":  0,
  "readOnly":  false,
  "dataCenter":  "",
  "rack":  "",
  "dataNode":  "",
  "maxFileNameLength":  0
},
```

Now, I want to implement something like writie cache using volume.tier.move: I want to store the hot data in **ssd** (which has its on redundancy), **without any replication**, while store the cold data using **hdd**, **with replication 001**
How can I achieve that?

**Comment by chrislusf** [comment]
You can just configure the hot data to ssd. And run scripts periodically to move volumes to hdd.
However, current code does not support different replication, which is technically possible, but no generic good way to specify the replication level. Need some ideas.

**Reply by NyaMisty** [reply, marked as answer]
actually I've found the solution my self.
It's not possible to configure replication based on node, because there's no exposed interfaces for that.
But actually the replication is configured on volume level, and the corresponding command for that is `volume.configure.replication`.
Based on source code I'm lucky to find `volume.tier.move` has a `-toReplication` option, which is exactly what I need.
All I need to do is something like:
`volume.tier.move -force -collectionPattern XXXX -fromDiskType ssd -toDiskType hdd -toReplication 001`
